### PR TITLE
Diff with clipboard support

### PR DESCRIPTION
**Pull request by pbaldauf**
This is the plugin I was always looking for since moving from Sublime Text to VScode.

The only thing I am really missing is the "Diff with clipboard" feature provided by the **FileDiffs Plugin** for Sublime (https://github.com/colinta/SublimeFileDiffs)
Think it would be even possible to support additional external diff tools, but that's a feature for the future ;)

This PR adds support for "Diff with clipboard" in same fashion as for Sublime.
image

image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -51,6 +52,11 @@
 				"title": "Current file to one other file (not open in editor)",
 				"category": "Meld Diff"
 			},
+			{
+				"command": "meld-diff.diffWithClipboard",
+				"title": "Diff with clipboard",
```
Comment (danielroedl) [inline package.json:57]: I would prefer according to `meld-diff.diffFromFileList` to use `"Compare with clipboard"` as title.

**Comment by danielroedl** [comment]
Thanks for the PR. I don't know this option of the FileDiffs Plugin, but it's very useful!

I just have made one comment regarding the title of the new command. I think the rest is ok.

I like the way how you compare the selection with usage of a temp file. I think I will add based on this also an option to compare not saved files in the same way (or changed files that are not yet saved).
Secondly I will also add the settings options to configure the tool used for the diff. This change I also want to make a long time. Thanks for the reminder. :smile: 

I hope I will found some time this weekend for these two changes. Maybe I can release them together with your "clipboard command" change.

File: package.json (hunk 1 above)
Comment (pbaldauf) [reply package.json]: Thanks for your constructive feedback. Renaming the title is also ok for me.

I really like the idea of supporting additional diff tools. Using settings options is definitively a good idea.

**Comment by danielroedl** [comment]
Thanks for the support!

File: src/extension.ts [hunk 2]
```diff
@@ -138,7 +176,47 @@ export function activate(context: vscode.ExtensionContext) {
...
+			const editorContent = editor.document.getText(selection);
+			current = await writeFileOnDisk(editorContent);
+			sameContent = await areFilesEqual([current, clipboard]);
+		} else if (editor.document.isUntitled) {
+			//compare against untitled file
+			const editorContent = editor.document.getText();
+			current = await writeFileOnDisk(editorContent);
+			sameContent = await areFilesEqual([current, clipboard]);
+		} else if (editor.document.isDirty) {
+			//compore against dirty content but invoke meld with current saved file
```
Comment (danielroedl) [inline src/extension.ts:204]: I just saw it now as I developed on top of your changes.
Why do you compare the dirty content but invoke meld with the saved file? Is this also the behavior of the sublime extension?

For me this behavior is confusing.

If I run the command I would expect either to check **and** open the dirty version or check **and** open the saved version of the file.

Background of this question is that I want to add a settings value for the extension were you can select if generally the dirty version or the saved version should be used to compare. So I would make this also equal in the compare to clipboard function.

**Pull request merged**